src/log/io_backend.cr

```diff
@@ -0,0 +1,46 @@
+# A `Log::Backend` that emits to an `IO` (defaults to STDOUT).
+class Log::IOBackend < Log::Backend
+  property io : IO
+  property progname : String
+  property formatter : Formatter? = nil
+
+  def initialize(@io = STDOUT)
+    @mutex = Mutex.new(:unchecked)
+    @progname = File.basename(PROGRAM_NAME)
+  end
+
+  def write(entry : Entry)
+    @mutex.synchronize do
+      format(entry)
+      io.puts
+      io.flush
+    end
+  end
+
+  # Emits the *entry* to the given *io*.
+  # It will use the `#formatter` if defined, otherwise will call `#default_format`.
+  def format(entry : Entry)
+    if formatter = @formatter
+      formatter.call(entry, io)
+    else
+      default_format(entry)
+    end
+  end
+
+  # Emits the *entry* to the given *io*.
+  def default_format(entry : Entry)
+    label = entry.severity.label
+    io << label[0] << ", ["
+    entry.timestamp.to_rfc3339(io)
+    io << " #" << Process.pid << "] "
+    io << label.rjust(7) << " -- " << @progname << ":" << entry.source << ": " << entry.message
```

Comment:
This `label.rjust(7)` is the only bit that allocates memory when logging info.

Suggestion: just like we have a `label` method in `Severity`, introduce a new method `label_rjust7` or similar with the labels with necessary spaces. Maybe not make it public (nodoc), but it can be useful for other formatters too.

Then the performance is a bit better, but mainly there are no allocations which is really good for a logging system.

Reply:
Actually, nevermind. It would be nice to have `rjust`, `ljust` and `center` write to an IO too. I'll try to implement that.

Reply:
Done! #8923 is merged so you can rebase against master and use that io overload.

Reply:
done 👍